src/ci/run.sh

```diff
@@ -67,6 +67,12 @@ else
   fi
 fi
 
+# We've had problems in the past of shell scripts leaking fds into the sccache
+# server (#48192) which causes Cargo to erroneously think that a build script
+# hasn't finished yet. Try to solve that problem by starting a very long-lived
+# sccache server at the start of the build, but no need to worry if this fails.
+SCCACHE_IDLE_TIMEOUT=7200 sccache --start-server || true
```

Comment:
This seems insufficient -- 120 minutes is only 2 hours, whereas our builders go up to 3 hours sometimes. Maybe we should lengthen it just to make sure?

Reply:
Seems fine by me!